annotation-processor/src/main/java/org/fulib/fx/FulibFxProcessor.java

```diff
@@ -273,8 +294,68 @@ private boolean isController(TypeMirror typeMirror) {
-        return typeMirror.toString().startsWith("javax.inject.Provider");
+    /**
+     * Checks if an element is an event element (annotated with {@link ControllerUtil#EVENT_ANNOTATIONS}).
+     *
+     * @param element The element to check
+     * @return True if the element is an event element
+     */
+    private boolean isEventElement(Element element) {
+        for (Class<? extends Annotation> eventAnnotation : ControllerUtil.EVENT_ANNOTATIONS) {
+            if (element.getAnnotation(eventAnnotation) != null) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Checks if the given method overrides another event method.
+     *
+     * @param method       The method to check
+     * @param eventMethods A map of all event methods of the superclasses
+     */
+    private void checkOverrides(ExecutableElement method, Map<String, List<ExecutableElement>> eventMethods) {
+        TypeElement element = (TypeElement) processingEnv.getTypeUtils().asElement(method.getEnclosingElement().asType());
+
+        // If no parent class is found, the method cannot override anything
+        if (element.getSuperclass().getKind() == TypeKind.NONE) {
+            return;
+        }
+
+        // Check if one of the parent classes has the method
+        if (eventMethods.containsKey(method.getSimpleName().toString())) {
+            for (ExecutableElement parentMethod : eventMethods.get(method.getSimpleName().toString())) {
+                if (sameMethodSignature(method, parentMethod)) {
+                    Name className = element.getQualifiedName();
+                    Name parentClassName = ((TypeElement) parentMethod.getEnclosingElement()).getQualifiedName();
+                    String error = error(1013).formatted(method, className, parentClassName);
+                    processingEnv.getMessager().printMessage(Diagnostic.Kind.ERROR, error, method);
+                    break;
+                }
+            }
+        }
+    }
+
+    private boolean sameMethodSignature(ExecutableElement method, ExecutableElement otherMethod) {
+        ExecutableType methodType = (ExecutableType) method.asType();
+        ExecutableType otherMethodType = (ExecutableType) otherMethod.asType();
+        return method.getSimpleName().equals(otherMethod.getSimpleName()) &&
+            method.getParameters().size() == otherMethod.getParameters().size() &&
+            processingEnv.getTypeUtils().isSubsignature(methodType, otherMethodType);
+    }
+
+    private void checkEventModifiers(TypeElement clazz) {
+        Stream.concat(helper.streamAllMethods(clazz), helper.streamAllFields(clazz))
+            .filter(this::isEventElement)
+            .filter(element -> element.getModifiers().contains(Modifier.PRIVATE))
+            .forEach(element -> {
+                String kind = element.getKind() == ElementKind.METHOD ? Method.class.getSimpleName() : Field.class.getSimpleName();
```

Comment:
perhaps toLowerCase to not seem out of place? or just use string literals, its not like this is gonna change ^^